libs/wire-subsystems/src/Wire/ConversationStore/Postgres.hs

```diff
@@ -0,0 +1,1228 @@
+                            |]
+
+setSelfMemberRemoteConv :: (PGConstraints r) => UserId -> Remote ConvId -> MemberUpdate -> Sem r ()
+setSelfMemberRemoteConv uid (tUntagged -> Qualified cid domain) MemberUpdate {..} =
+  runStatement
+    (uid, domain, cid, mupOtrMuteStatus, mupOtrMuteRef, mupOtrArchive, mupOtrArchiveRef, mupHidden, mupHiddenRef)
+    update
+  where
+    update :: Hasql.Statement (UserId, Domain, ConvId, Maybe MutedStatus, Maybe Text, Maybe Bool, Maybe Text, Maybe Bool, Maybe Text) ()
+    update =
+      lmapPG
+        [resultlessStatement|UPDATE remote_conversation_local_member
+                             SET otr_muted_status = COALESCE($4 :: integer?, otr_muted_status),
+                                 otr_muted_ref =    COALESCE($5 :: text?,    otr_muted_ref),
+                                 otr_archived =     COALESCE($6 :: boolean?, otr_archived),
+                                 otr_archived_ref = COALESCE($7 :: text?,    otr_archived_ref),
+                                 hidden =           COALESCE($8 :: boolean?, hidden),
+                                 hidden_ref =       COALESCE($9 :: text?,    hidden_ref)
+                             WHERE "user" = ($1 :: uuid)
+                             AND conv_remote_domain = ($2 :: text)
+                             AND conv_remote_id = ($3 :: uuid)
+                            |]
+
+setOtherMemberImpl :: (PGConstraints r) => Local ConvId -> Qualified UserId -> OtherMemberUpdate -> Sem r ()
+setOtherMemberImpl lcnv =
+  foldQualified
+    lcnv
+    (setOtherLocalMember (tUnqualified lcnv) . tUnqualified)
+    (setOtherRemoteMember (tUnqualified lcnv))
+
+setOtherLocalMember :: (PGConstraints r) => ConvId -> UserId -> OtherMemberUpdate -> Sem r ()
+setOtherLocalMember cid uid upd =
+  for_ upd.omuConvRoleName $ \newRole ->
+    runStatement (cid, uid, newRole) update
+  where
+    update :: Hasql.Statement (ConvId, UserId, RoleName) ()
+    update =
+      lmapPG
+        [resultlessStatement|UPDATE conversation_member
+                             SET conversation_role = ($3 :: text)
+                             WHERE conv = ($1 :: uuid)
+                             AND "user" = ($2 :: uuid)
+                            |]
+
+setOtherRemoteMember :: (PGConstraints r) => ConvId -> Remote UserId -> OtherMemberUpdate -> Sem r ()
+setOtherRemoteMember cid (tUntagged -> Qualified uid domain) upd =
+  for_ upd.omuConvRoleName $ \newRole ->
+    runStatement (cid, domain, uid, newRole) update
+  where
+    update :: Hasql.Statement (ConvId, Domain, UserId, RoleName) ()
+    update =
+      lmapPG
+        [resultlessStatement|UPDATE local_conversation_remote_member
+                             SET conversation_role = ($4 :: text)
+                             WHERE conv = ($1 :: uuid)
+                             AND user_remote_domain = ($2 :: text)
+                             AND user_remote_id = ($3 :: uuid)
+                            |]
```

Comment:
implement in terms of `setSelfMemberImpl`?

Reply:
The `setSelfMemberImpl` doesn't update the role, and this only updates the role. Here the `foldQualified` is working on whether or not the member is remote, while the self case folds over whether or not the conv is remote.  So I think its not worth trying to unify these.